ProcFilter.cpp

```diff
@@ -146,65 +59,206 @@ bool ProcFilter::ParseConfig(const Config& config) {
+                            _filters.clear();
+                            return false;
+                        }
+                        match_mask |= PFS_MATCH_UID;
                     } else {
                         Logger::Error("Invalid entry (%s) at (%d) in config for '%s'", mi->name.GetString(), idx, CONFIG_PARAM_NAME.c_str());
                         _filters.clear();
                         return false;
                     }
-                } else {
-                    Logger::Error("Invalid entry (%s) at (%d) in config for '%s' is missing", mi->name.GetString(), idx, CONFIG_PARAM_NAME.c_str());
-                    _filters.clear();
-                    return false;
                 }
-                mi = it->FindMember("arg1_prefix");
+
+                mi = it->FindMember("group");
                 if (mi != it->MemberEnd()) {
                     if (mi->value.IsString()) {
-                        arg1 = std::string(mi->value.GetString(), mi->value.GetStringLength());
+                        group = std::string(mi->value.GetString(), mi->value.GetStringLength());
+                        if (is_number(group)) {
+                            gid = std::stoi(group);
+                        } else {
+                            gid = _user_db->GroupNameToGid(group);
+                        }
+                        if (gid == -1) {
+                            Logger::Error("Invalid entry (%s) at (%d) in config for '%s'", mi->name.GetString(), idx, CONFIG_PARAM_NAME.c_str());
+                            _filters.clear();
+                            return false;
+                        }
+                        match_mask |= PFS_MATCH_GID;
                     } else {
                         Logger::Error("Invalid entry (%s) at (%d) in config for '%s'", mi->name.GetString(), idx, CONFIG_PARAM_NAME.c_str());
                         _filters.clear();
                         return false;
                     }
                 }
-                mi = it->FindMember("flags");
+
+                mi = it->FindMember("syscalls");
                 if (mi != it->MemberEnd()) {
-                    if (mi->value.IsInt()) {
-                        int i = mi->value.GetInt();
-                        if (i <= 0 || i > 0xFFFF) {
+                    if (mi->value.IsArray()) {
+                        if (!syscalls.empty()) {
+                            syscalls.clear();
+                        }
+                        bool includesExclude = false;
+                        for (auto it2 = mi->value.Begin(); it2 != mi->value.End(); ++it2) {
+                            syscalls.emplace_back(std::string(it2->GetString(), it2->GetStringLength()));
+                            if (it2->GetString()[0] == '!') {
+                                includesExclude = true;
+                            }
+                        }
+                        if (includesExclude) {
```

Comment:
Didn't really follow what this is for?

Reply:
Yes. I toyed with variations and decided this made most sense. If the syscall list only has includes (no exclamation marks), then we filter out those and alert on everything else. If the syscall list only has excludes (with exclamation marks) then we alert on the excludes and filter everything else. If it is a mixture then it's hard to say what the default action should be so I made the decision to make it default exclude. We can discuss and change this in the future if a use case can show that this makes less sense than an alternative.

Reply:
I have done in my own testing.

Reply:
Thanks @taglines. I changed functionality to your suggestion.